File: frontend/src/api/applications.js
import client from "./client";

/**
 * POST /applications/
 * body: { customer_id, loan_amount, loan_term, employment_years }
 * returns: ApplicationResponse { id, status, ...body }
 */
export function createApplication(data) {
  return client.post("/applications/", data).then((res) => res.data);
}

/**
 * GET /applications/{application_id}
 */
export function getApplication(applicationId) {
  return client
    .get(`/applications/${applicationId}`)
    .then((res) => res.data);
}

File: backend/app/api/applications.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from app.database.connection import get_db
from app.schemas.application import (
    ApplicationCreate,
    ApplicationResponse
)
from app.services.application_service import (
    ApplicationService
)


router = APIRouter(
    prefix="/applications",
    tags=["Applications"]
)


@router.post(
    "/",
    response_model=ApplicationResponse
)
def create_application(
    data: ApplicationCreate,
    db: Session = Depends(get_db)
):

    return ApplicationService.create(
        db,
        data
    )


@router.get("/{application_id}")
def get_application(
    application_id: int,
    db: Session = Depends(get_db)
):

    application = ApplicationService.get(
        db,
        application_id
    )

    if not application:
        raise HTTPException(
            status_code=404,
            detail="Application not found"
        )

    return application